pytest_datadir/plugin.py

```diff
@@ -25,24 +26,106 @@ def _win32_longpath(path):
         return path
 
 
-@pytest.fixture
-def shared_datadir(request, tmpdir):
-    original_shared_path = os.path.join(request.fspath.dirname, 'data')
-    temp_path = Path(str(tmpdir.join('data')))
-    shutil.copytree(_win32_longpath(original_shared_path), _win32_longpath(str(temp_path)))
-    return temp_path
-
-
 @pytest.fixture
 def original_datadir(request):
     return Path(os.path.splitext(request.module.__file__)[0])
 
 
-@pytest.fixture
-def datadir(original_datadir, tmpdir):
-    result = Path(str(tmpdir.join(original_datadir.stem)))
-    if original_datadir.is_dir():
-        shutil.copytree(_win32_longpath(str(original_datadir)), _win32_longpath(str(result)))
-    else:
-        result.mkdir()
-    return result
+class DatadirFactory(object):
+    """Factory class for generating datadir fixtures."""
+
+    def __init__(self, request, tmp_path_factory):
+
+        self.tmp_path_factory = tmp_path_factory
+        self.request = request
+
+    def mkdatadir(self, original_datadir='data'):
```

Comment:
Hmm seems strange to me to have `'data'` as default. I would expect this to be `None`. Do you have a use-case?

Reply:
My thought process: I was thinking that since this was a hardcoded behavior in the past that if someone were to want to replicate that behavior they would have to look up the specific string for backwards compatibility, so I thought it was a good place to put it (there should also be a module or class constant as well). I view `None` (and secondarily `Ellipsis`) as sort of the other obvious defaults since they only signify themselves and would have to be interpreted by special logic in the module to mean anything. For strings the user is free to make their own labels however there is one semi-special string, 'data', that is a module specific convention and the user looking at the autodocs signature or docstring and seeing that as the default would recognize that.

I think your judgement is best since I am really brand new to this sphere.